daemon/state.go

```diff
@@ -304,12 +304,18 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState) (resto
 			// the identity even if has not changed.
 			ep.SetIdentity(identity)
 
+			// We don't need to hold the policy repository mutex here because
+			// the content of the rules themselves are not being changed.
+			wg := d.policy.UpdateLocalConsumers([]policy.IdentityConsumer{ep})
```

Comment:
Might not be necessary to invoke this here since lazy-evaluation within `rule.matches` always will update the cache anyway; thoughts? The nice thing about this is that it's done outside of the `regenerate` logic, which holds one of the builders for endpoints, so once regeneration is triggered for an endpoint in the future which holds a builder, evaluation wouldn't take as long since the caches in the rules are already populated.

Reply:
Yes, we ensure that all policy is received from K8s before restoring, as otherwise we would have an incomplete view of what policy is in relation to all endpoints on the node. This is done in `Daemon.initK8sSubsystem` before `daemon.regenerateRestoredEndpoints` in the `runDaemon` function in `daemon/`.